api/v1/manager_types.go

```diff
@@ -72,8 +68,8 @@ type ManagerDeploymentPodSpec struct {
 // ManagerDeploymentContainer is a Manager Deployment container.
 type ManagerDeploymentContainer struct {
 	// Name is an enum which identifies the Manager Deployment container by name.
-	// Supported values are: tigera-voltron, tigera-manager, tigera-es-proxy
-	// +kubebuilder:validation:Enum=tigera-voltron;tigera-manager;tigera-es-proxy
+	// Supported values are: tigera-voltron, tigera-manager, tigera-ui-apis, and tigera-es-proxy (deprecated).
+	// +kubebuilder:validation:Enum=tigera-voltron;tigera-manager;tigera-es-proxy;tigera-ui-apis
 	Name string `json:"name"`
```

Comment:
what happens if value passed in is `tigera-es-proxy` since it's still a valid input?

Reply:
Yeah, this is an interesting one - It will be accepted but not do anything.

I think the upgrade should be that you create duplicate entries - one for each container - then upgrade, and then remove the old `tigera-es-proxy` entry.